tools/wptrunner/wptrunner/executors/executorselenium.py

```diff
@@ -193,12 +196,50 @@ def do_test(self, test):
         return (test.result_cls(*data), [])
 
     def do_testharness(self, webdriver, url, timeout):
-        return webdriver.execute_async_script(
-            self.script % {"abs_url": url,
-                           "url": strip_server(url),
-                           "window_id": self.window_id,
-                           "timeout_multiplier": self.timeout_multiplier,
-                           "timeout": timeout * 1000})
+        format_map = {"abs_url": url,
+                      "url": strip_server(url),
+                      "window_id": self.window_id,
+                      "timeout_multiplier": self.timeout_multiplier,
+                      "timeout": timeout * 1000}
+        webdriver.execute_script(self.script % format_map)
+        while True:
+            result = webdriver.execute_async_script(
+                self.script_resume % format_map)
+            if result[1] == "complete":
+                result = [result[0]] + result[2]
+                break
+            elif result[1] == "action":
+                parent = webdriver.current_window_handle
+                try:
+                    webdriver.switch_to.window(self.window_id)
+                    action = result[2]["action"]
+                    if action == "click":
+                        selector = result[2]["selector"]
+                        elements = webdriver.find_elements_by_css_selector(selector)
+                        assert len(elements) == 1  # the JS should ensure this
+                        try:
+                            elements[0].click()
+                        except (exceptions.ElementNotInteractableException,
+                                exceptions.ElementNotVisibleException) as e:
+                            msg = json.dumps(
+                                {
+                                    "type": "testautomation-complete",
+                                    "status": "failure",
+                                    "message": str(e)
+                                })
+                            webdriver.execute_script("window.postMessage(%s, '*')" % msg)
+                        else:
+                            msg = json.dumps(
+                                {
+                                    "type": "testautomation-complete",
+                                    "status": "success"
+                                })
+                            webdriver.execute_script("window.postMessage(%s, '*')" % msg)
+                finally:
+                    import time; time.sleep(1)
```

Comment:
This is needed for the fullscreen test to run to completion. Seemingly, if we change away from the window too soon after doing the click the browser never has time to enter fullscreen (and doesn't unless it's the window with focus) and therefore the test just times out.

cc/ @jgraham @foolip